app/credentials/concerns/user_provided_services.rb

```diff
@@ -0,0 +1,17 @@
+module UserProvidedService
+  def credentials(service_name)
+    user_provided_service[service_name]
+  end
+
+  def use_env_var?
+    Rails.env.development? || Rails.env.test?
+  end
+
+  def user_provided_service
+    vcap_services["user-provided"][0]["credentials"]
```

Comment:
This assumes that there's only going to be one UPS, and it will hold all of the credentials. The impression I get is that that's not how UPSes should be used - rather, a single UPS per external service, so we'd have one for Mandrill, one for cloud.gov auth, etc. (This would also make the credentials much easier to edit.)